### PR TITLE
Use pytest-httpbin to run the test suite locally

### PR DESCRIPTION
**Pull request by mgorny**
Replace the use of online httpbin with the local httpbin instance
provided via pytest-httpbin package.  This makes the test suite offline,
effectively removing the need for pytest-network as well.

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -1,34 +1,60 @@
...
-rich==13.4.2
+rich==13.5.2
+rpds-py==0.9.2
 six==1.16.0
 twine==4.0.2
 webencodings==0.5.1
-zipp==3.15.0
+zipp==3.16.2
+zope.event==5.0
+zope.interface==6.0
```
Comment (ross) [inline requirements-dev.txt:60]: These updated with `./script/update-requirements`? When I run it locally on this branch I get a few diffs:

```console
coho:requests-futures ross$ git diff
diff --git a/requirements-dev.txt b/requirements-dev.txt
index 330d5b6..c652801 100644
--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -3,7 +3,6 @@ Jinja2==3.1.2
 MarkupSafe==2.1.3
 PyYAML==6.0.1
 Pygments==2.16.1
-SecretStorage==3.3.3
 Werkzeug==2.3.6
 attrs==23.1.0
 black==23.7.0
@@ -14,7 +13,6 @@ build==0.10.0
 cffi==1.15.1
 click==8.1.6
 coverage==7.2.7
-cryptography==41.0.3
 decorator==5.1.1
 docutils==0.20.1
 flasgger==0.9.7.1
@@ -26,7 +24,6 @@ iniconfig==2.0.0
 isort==5.12.0
 itsdangerous==2.1.2
 jaraco.classes==3.3.0
-jeepney==0.8.0
 jsonschema-specifications==2023.7.1
 jsonschema==4.19.0
 keyring==24.2.0
```

**Review by ross — COMMENTED** [review]
Looks reasonable. One question inline before ✅

File: requirements-dev.txt (hunk 1 above)
Comment (ross) [reply requirements-dev.txt:60]: Doesn't appear to be:

```console
(env) coho:requests-futures ross$ python --version
Python 3.11.4
```

Maybe OS. I'm on OSX. Anyway. Just seemed like odd requirements, but probably not hurting anything, I guess.

**Review by ross — APPROVED** [review]

File: requirements-dev.txt (hunk 1 above)
Comment (mgorny) [reply requirements-dev.txt:60]: Ah yes, it's probably because I'm running on Linux ;-).

**Comment by mgorny** [comment]
Thanks!

**Comment by ross** [comment]
Thank you for the investigation/PR.

**Pull request merged**